### PR TITLE
8.0 base location geonames import

### PR DESCRIPTION
**Pull request by eLBati**
Starting from @alexis-via [MP](https://code.launchpad.net/~akretion-team/partner-contact-management/base-location-geonames-import/+merge/214564) and porting to 8.0
- using new API
- adding tests
- refactoring with `select_or_create_state` in order to [create states if they don't exist](https://code.launchpad.net/~akretion-team/partner-contact-management/base-location-geonames-import/+merge/214564/comments/538000)

The new `select_or_create_state` also allows to be extended using different CSV fields for states

### CODE REVIEW

**Comment by dcorio** [comment]
i got this while trying to run the import wizard for the second time:

  File "/Users/enlightx/Development/Odoo/8.0/dev/parts/8.0-base-location-geonames-import/base_location_geonames_import/wizard/geonames_import.py", line 109, in run_import
    bzip_obj.unlink(bzip_ids_to_delete)
  File "/Users/enlightx/Development/Odoo/8.0/dev/parts/odoo/openerp/api.py", line 231, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/Users/enlightx/Development/Odoo/8.0/dev/parts/odoo/openerp/api.py", line 548, in new_api
    result = method(self._model, cr, uid, self.ids, _args, *_kwargs)
TypeError: unlink() got multiple values for keyword argument 'context'

**Comment by eLBati** [comment]
Thanks @dcorio
I added the test case and fixed the bug

File: base_location_geonames_import/wizard/geonames_import.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                'code': row[code_row_index],
+                'country_id': country_id
+                })
+
+    @api.one
+    def run_import(self):
+        bzip_obj = self.env['res.better.zip']
+        country_id = self.country_id.id
+        country_code = self.country_id.code.upper()
+        url = 'http://download.geonames.org/export/zip/%s.zip' % country_code
```
Comment (hbrunn) [inline base_location_geonames_import/wizard/geonames_import.py]: wouldn't it make sense to put this into an ir.config_parameter? That would make eventual changes in the URL a lot simpler to apply for non-technical admins and also wouldn't require a server restart

**Comment by hbrunn** [comment]
:+1:

File: base_location_geonames_import/wizard/geonames_import.py (hunk 1 above)
Comment (eLBati) [inline base_location_geonames_import/wizard/geonames_import.py]: It does.
Just commited https://github.com/eLBati/partner-contact/commit/9e426a0226d042099b6d60d6a3b57cc5f393c9ce
Thanks @hbrunn

**Comment by FrancoTampieri** [comment]
+1 for the merge, I tested the import without problem

**Pull request merged**